common/app/routes/Challenges/Child-Container.jsx

```diff
@@ -1,20 +1,57 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import { challengeSelector } from '../../redux';
+import { challengeUpdated } from './redux';
 
 import CompletionModal from './Completion-Modal.jsx';
 import AppChildContainer from '../../Child-Container.jsx';
+import { OverlayLoader } from '../../helperComponents';
+import { fullBlocksSelector } from '../../entities';
+import { paramsSelector } from '../../Router/redux';
+
+const mapStateToProps = createSelector(
+  challengeSelector,
+  fullBlocksSelector,
+  paramsSelector,
+  (challenge, fullBlocks, { block }) => ({
+      challenge,
+      showLoading: !fullBlocks.includes(block)
```

Comment:
@BerkeleyTrue this should be more robust

Reply:
:+1: